### ISSUE TITLE
Crash on load when rocket-part recipe is removed

### ISSUE THREAD

**Issue by PFQNiet** [Helfima/helmod#353]
My overhaul mod includes deleting the `rocket-part` recipe from the base game. Helmod apparently has this recipe hardcoded, and this leads to an error when starting a game:

```
The mod Helmod: assistant for planning your base. (0.12.5) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event helmod::on_init()
__helmod__/model/Player.lua:873: attempt to index local 'rocket_part_prototype' (a nil value)
stack traceback:
	__helmod__/model/Player.lua:873: in function 'getRecipeRocket'
	__helmod__/model/RecipePrototype.lua:35: in function 'init'
	__helmod__/core/class.lua:25: in function <__helmod__/core/class.lua:21>
	(...tail calls...)
	__helmod__/selector/RecipeSelector.lua:62: in function 'appendGroups'
	__helmod__/selector/RecipeSelector.lua:110: in function 'updateGroups'
	__helmod__/selector/AbstractSelector.lua:403: in function 'prepare'
	__helmod__/controller/Controller.lua:136: in function 'on_init'
	__helmod__/controller/EventController.lua:108: in function <__helmod__/controller/EventController.lua:106>
```

I've "fixed" this on my end by adding a dummy `rocket-part` hidden recipe, but I'm reporting this here for potential fixing.

**Comment by 88react** [comment]
As far as I understand, your mod completely removes the recipes related to the launch of the rocket (rocket-silo, satellite, rocket-part) and does not use this mechanic at all. But due to the fact that you only hide, but do not delete items associated with this mechanic, this error occurs. You can avoid this in several ways. 1 Do not delete these recipes. 2 Remove and items related to this mechanic (having a rocket_launch_product). 3 Remove rocket_launch_product from all items.

**Comment by PFQNiet** [comment]
The function `getRecipeRocket` calls upon the `rocket-part` recipe directly by name, and does not verify that the recipe exists. That's what is causing this error.

**Comment by Helfima** [comment]
fixed in v0.12.6
